File: baekjoon/baekjoon_10845.cpp
#include <bits/stdc++.h>

using namespace std;
int main(void) {
    ios::sync_with_stdio;
    cin.tie(0);
    int N;
    string command;
    queue<int> Q;
    int X;
    cin >> N;
    for (int i = 0; i < N; i++) {
        cin >> command;
        if (command == "push") {
            cin >> X;
            Q.push(X);
        } else if (command == "pop") {
            if (Q.empty()) {
                cout << -1 << "\n";
            } else {
                cout << Q.front() << "\n";
                Q.pop();
            }
        } else if (command == "size") {
            cout << Q.size() << "\n";
        } else if (command == "empty") {
            if (Q.empty()) {
                cout << 1 << "\n";
            } else {
                cout << 0 << "\n";
            }
        } else if (command == "front") {
            if (Q.empty()) {
                cout << -1 << "\n";
            } else {
                cout << Q.front() << "\n";
            }
        } else if (command == "back") {
            if (Q.empty()) {
                cout << -1 << "\n";
            } else {
                cout << Q.back() << "\n";
            }
        }
    }
}

File: baekjoon/baekjoon_10845.py
import sys
from collections import deque
N = int(sys.stdin.readline().rstrip())
q = deque()
for i in range(N):
    commands = sys.stdin.readline().rstrip().split()
    if commands[0]=="push":
        q.append(int(commands[1]))
    elif commands[0]=="pop":
        if q:
            print(q.popleft())
        else:
            print(-1)
    elif commands[0]=="size":
        print(len(q))
    elif commands[0]=="empty":
        if q:
            print(0)
        else:
            print(1)
    elif commands[0]=="front":
        if q:
            print(q[0])
        else:
            print(-1)
    elif commands[0]=="back":
        if q:
            print(q[-1])
        else:
            print(-1)